### PR TITLE
Add map component tests

### PR DESCRIPTION
**Pull request by tonypls**
## Issue
We want to refactor the Map.tsx file but it does not have sufficient test coverage

## Description
This PR adds a cypress component test for the map which intercepts and stubs the state api. 

The map is a canvas so it's unable to be selected by cypress in the dom. To work around this I've added an optional callback to the map component that can execute tests with context of the map and it's features.

### Double check

- [x] I have tested my parser changes locally with `poetry run test_parser "zone_key"`
- [x] I have run `pnpx prettier --write .` and `poetry run format` to format my changes.

### CODE REVIEW

**Comment by madsnedergaard** [comment]
Regarding the "any" warnings, I suggest we disable that rule for tests. In my opinion, it's better to have more tests than perfectly typed test - and there's a risk that we end up doing less because typing them up perfectly is extra work :)

So I suggest adding the following override to `.eslintrc.json`:
```json
{
  "files": ["src/**/*.{cy,stories,test}.ts?(x)"],
  "rules": {
    "@typescript-eslint/no-explicit-any": "off"
  }
},
```


cc @VIKTORVAV99, wdyt?

**Comment by VIKTORVAV99** [comment]
> Regarding the "any" warnings, I suggest we disable that rule for tests. In my opinion, it's better to have more tests than perfectly typed test - and there's a risk that we end up doing less because typing them up perfectly is extra work :)
> 
> 
> 
> So I suggest adding the following override to `.eslintrc.json`:
> 
> ```json
> 
> {
> 
>   "files": ["src/**/*.{cy,stories,test}.ts?(x)"],
> 
>   "rules": {
> 
>     "@typescript-eslint/no-explicit-any": "off"
> 
>   }
> 
> },
> 
> ```
> 
> 
> 
> 
> 
> cc @VIKTORVAV99, wdyt?

Haven't we already done so for a few rules? I don't think it's an issue if we expanded it to this as well.

Obviously it would be good if all code was 100% typed but the important part is what goes into the app (and server) not the tests.

**Comment by madsnedergaard** [comment]
> Haven't we already done so for a few rules? I don't think it's an issue if we expanded it to this as well.
> 
> Obviously it would be good if all code was 100% typed but the important part is what goes into the app (and server) not the tests.

Yup, so maybe we can align them further - I'll create a commit here :)

**Review by madsnedergaard — APPROVED** [review]
Oh sorry, thought I had submitted!
One tiny nitpick, but feel free to ignore it :)

File: web/src/features/map/Map.tsx [hunk 1]
```diff
@@ -36,10 +36,15 @@ const MAP_STYLE = {
   layers: [],
   glyphs: 'fonts/{fontstack}/{range}.pbf',
 };
-const isMobile = window.innerWidth < 768;
+const IS_MOBILE = window.innerWidth < 768;
```
[the remark this answers is not in the document]
Comment (tonypls) [reply web/src/features/map/Map.tsx]: Fair play :)

**Comment by madsnedergaard** [comment]
> ## Suspect Issues
> This pull request was deployed and Sentry observed the following issues:
> 
> * ‼️ **React ErrorBoundary Error: Style is not done loading.** `d(src/features/map/Map)` [View Issue](https://electricitymaps.sentry.io/issues/4642065730/?referrer=github-pr-bot)
> 
> Did you find this useful? React with a 👍 or 👎

Lol thanks Sentry, I don't quite think it was this PR though 😅

**Comment by VIKTORVAV99** [comment]
Poor Sentry is just trying to help out...

**Pull request merged**